packages/sdk/src/mb-impl/token.ts

```
@@ -0,0 +1,214 @@
+  if (network === Network.MAINNET) return MB_MAINNET_TOKEN_FACTORY;
+  else return MB_TESTNET_TOKEN_FACTORY;
+};
+
+// TODO: figure out a way to generate gas and deposit for each
+
+export const transfer = (args: TransferArgs): TransactionArgs => {
+  const { nftContractId, transfer } = args;
+
+  const isBatchTransfer = transfer.length > 1;
+
+  if (isBatchTransfer) {
+    const ids = transfer.map((transferElm) => {
+      return [transferElm.receiverId, transferElm.tokenId];
+    });
+
+    return {
+      contractAddress: nftContractId,
+      methodName: METHOD_NAMES.BATCH_TRANSFER,
+      args: {
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        token_ids: ids,
+      },
+    };
+  } else {
+    const receiverId = transfer[0].receiverId;
+    const tokenId = transfer[0].tokenId;
+
+    return {
+      contractAddress: nftContractId,
+      methodName: METHOD_NAMES.TRANSFER,
+      args: {
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        receiver_id: receiverId,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        token_id: tokenId,
+      },
+    };
+  }
+};
+
+export const burn = (args: BurnArgs): TransactionArgs => {
+  const { nftContractId, tokenIds } = args;
+
+  return {
+    contractAddress: nftContractId,
+    methodName: METHOD_NAMES.BATCH_BURN,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      token_ids: tokenIds,
+    },
+  };
+};
+
+export const deployContract = (
+  args: DeployTokenContractArgs,
+): TransactionArgs => {
+  const {
+    name,
+    factoryContractId,
+    network = Network.TESTNET,
+    ownerId,
+    metadata,
+  } = args;
+
+  const { symbol, icon, baseUri, reference, referenceHash } = metadata;
+
+  const data = {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    owner_id: ownerId,
+    metadata: {
+      spec: 'nft-1.0.0',
+      name: name.replace(/[^a-z0-9]+/gim, '').toLowerCase(),
```

Shouldn't we return an error instead of silently overwriting user input? I feel this would be very annoying to debug.